ts/Dashboard/EditMode/Fullscreen.ts

```diff
@@ -0,0 +1,148 @@
+import Dashboard from '../Dashboard.js';
+import U from '../../Core/Utilities.js';
+
+const { addEvent } = U;
+
+class Fullscreen {
+    /* *
+     *
+     *  Static Functions
+     *
+     * */
+
+    /**
+     * Prepares the dashboard class to support fullscreen.
+     * @param {typeof_Dashboard} DashboardClass
+     *        The dashboard class to decorate with fullscreen support.
+     */
+    public static compose(DashboardClass: Dashboard): void {
+        // Initialize fullscreen
+        DashboardClass.fullscreen = new Fullscreen(DashboardClass);
+    }
+
+    /* *
+    *
+    *  Static Properties
+    *
+    * */
+
+
+    /* *
+    *
+    *  Constructor
+    *
+    * */
+    constructor(DashboardClass: Dashboard) {
+        this.isOpen = false;
+        this.dashboard = DashboardClass;
+    }
+
+    /* *
+    *
+    *  Properties
+    *
+    * */
+
+    public dashboard: Dashboard;
+    public isOpen: boolean;
+    public unbindFullscreenEvent?: Function;
```

Comment:
Is it not too general? I mean a type: `Function`

Reply:
Any idea how we can make it more precise?

Reply:
I've found the `Event` type, used in a few places.

@bre1470 what do you think?

Reply:
`Event` type is the type is the object. Here could be 
```ts
  public unbindFullscreenEvent?: () => void;
```

Reply:
I would expect this to become protected or private, as consumers should use `open` / `close` functions and have the `isOpen` flag. Otherwise the typing is fine for me. Microsoft likes to see the `() => void` used, but `Function` is good enough as we do not care about the return type.